src/utils/flow.js

```diff
@@ -46,6 +46,21 @@ export function convertFlowObjectTypeAnnotation(
       };
     }
 
+    if (value.type === "ArrayTypeAnnotation" && value.elementType) {
+      const children = value.elementType.id && flowTypes[value.elementType.id.name]
+        ? flowTypes[value.elementType.id.name]
+        : value.elementType;
+
+      return {
+        ...obj,
+        [key]: {
+          type: "array",
+          nullable: property.optional,
+          children: children.properties ? convertFlowObjectTypeAnnotation(children, flowTypes) : null
```

Comment:
Ahh sorry I see what you mean.

Yeah convertFlowObjectTypeAnnotation() will need to be turned into convertFlowTypeAnnotation(). There are too many places assuming and requiring objects, for example checkPropsObjectTypeMatchesSchema() only works if the type is an inline ObjectType.

I'm good to merge this as a starting point. I'll look into removing the object type assumptions unless you want to do it?